brave/brave-ios
# Fix #3397: Add ability to search bookmarks by title or URL in the bookmarks list

Adding Search Bar and functionality to Bookmark View Controller

## Summary of Changes

This pull request fixes #3397

## Submitter Checklist:


## Test Plan:

As indicated in the ticket, the search is global for all bookmarks. So whichever screen you are in the search query should bring results up to 200 among all the bookmarks.

The folders should not be shown in the result and when search br clicked empty query will bring empty results. If there is no results from the query empty search results screen should be presented. Edit and delete actions should work on searched objects and should effect corresponding object.

- Open Bookmarks, Click Search Bar
- Search a keyword and check If desired search results are listed 
- Swipe the table row element and edit the screen name and press save
- Check desired element name is updated.
- Write a non existing search keyword and check No Results found is displayed.
- Press cancel and check If we navigated back to originated controller.


## Screenshots:


https://user-images.githubusercontent.com/6643505/139488989-c3b01e3c-067b-417d-a3cb-88f0ab959ebe.MP4

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
